presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1607,6 +1668,59 @@ private void executeMetastoreDeleteOperations()
                 throw prestoException;
             }
         }
+
+        public ConnectorCommitHandle getCommitHandle()
+        {
+            Map<String, List<DateTime>> lastDataCommitTimes = new HashMap<>();
```

Comment:
use `ImmutableMap.Builder`. Check `modifiedPartitionMapBuilder` as an example in the class